django_saml2_auth/views.py

```diff
@@ -167,6 +185,24 @@ def acs(r):
     if authn_response is None:
         return HttpResponseRedirect(get_reverse([denied, 'denied', 'django_saml2_auth:denied']))
 
+    # must have all IDPs configure this value to return principal's email
+    user_name_id = None
+    try:
+        user_subject = authn_response.get_subject()
+        user_name_id = user_subject.text
+    except:
+        return HttpResponseRedirect(
+            get_reverse([denied, "denied", "django_saml2_auth:denied"])
+        )
+
+    # NOTE: to protect against exploit caused by malicious config of other idps
+    user_email_domain = user_name_id.split("@")[1]
+    if not expected_email_domain == user_email_domain:
```

Comment:
This ensures that the metadata_id matches the user you're trying to log in as. Does anything ensure that request is allowed to use this metadata_id? I assume the answer is yes but just to clarify what I'm asking. Is the following possible?

metadata_id_1 with the domain example1.org from IdP1
metadata_id_2 with the domain example2.org from IdP2

I send a request from IdP1 with metadata_id_2 in the url, and alex@example2.org as the user. Will I be able to log in?

Reply:
Confirmed that this exploit does not work on 12/29. The SAML Request AssertionConsumerServiceURL becomes invalid when there is a difference between the expected acs url generated in `signin()` vs `acs()`